tema.js

```diff
@@ -0,0 +1,164 @@
+'use strict'
+
+class erori extends Error {
+  static ERROR_CODE = {
+       METHOD_NOT_IMPLEMENTED: 'METHOD_NOT_IMPLEMENTED',
+       EXPIRED_PRODUCT: 'EXPIRED_PRODUCT',
+       NOT_ENOUGH_MONEY: 'NOT_ENOUGH_MONEY'
+   };
+
+   static ERROR_MESSAGE = {
+       [erori.ERROR_CODE.METHOD_NOT_IMPLEMENTED]: 'You must implement this method',
+       [erori.ERROR_CODE.EXPIRED_PRODUCT]: 'This product is expired',
+       [erori.ERROR_CODE.NOT_ENOUGH_MONEY]: "This customer doesn't have enough money"
+   };
+
+   constructor({ code, message = erori.ERROR_MESSAGE[code], data }) {
+       super(message);
+
+       this.code = code;
+       this.data = data;
+   }
+}
+
+
+class Person {
+  constructor() {}
+}
+
+class Customer extends Person {
+  #money
+  #shopping_cart
+
+  constructor (money, shopping_cart) {
+    super();
+    this.#money = money
+    this.#shopping_cart = shopping_cart
+  }
+
+  addToCart(product) {
+    try {
+      const data = product.getDate()
+      if(data - new Date() < 0)
+          throw new erori({
+                  code: erori.ERROR_CODE.EXPIRED_PRODUCT,
+                  data: { class: Customer.name }
+              });
+      this.#shopping_cart.push(product)
+    }
+    catch (error) {
+      if(error.code != 'EXPIRED_PRODUCT') {
+          this.#shopping_cart.push(product)
+      }
+      else {
+        throw new erori({
+                code: erori.ERROR_CODE.EXPIRED_PRODUCT,
+                data: { class: Customer.name }
+            });
+      }
+    }
+  }
+
+  removeFromCart(product) {
+    this.#shopping_cart.filter(crt => crt != product)
+  }
+
+  getShoppingCart() {return this.#shopping_cart;}
```

Comment:
It's not a good idea to return a private reference. It makes the whole concept of `private` pointless since whoever use this "getter" gets access to the internal state of the Customer instance and can mess with it.